staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go

```diff
@@ -678,18 +678,27 @@ func TestPatch(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	resourceVersion := createdNoxuInstance.GetResourceVersion()
 
 	// a patch with no change
-	createdNoxuInstance, err = noxuNamespacedResourceClient.Patch("foo", types.MergePatchType, patch, metav1.UpdateOptions{})
+	patchedNoxuInstance, err := noxuNamespacedResourceClient.Patch("foo", types.MergePatchType, patch, metav1.UpdateOptions{})
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	// make sure no-op patch does not increment resourceVersion
+	if patchedNoxuInstance.GetResourceVersion() != resourceVersion {
+		t.Fatalf("unexpected error: resourceVersion changed on no-op patch. expected %s, got %s", resourceVersion, patchedNoxuInstance.GetResourceVersion())
+	}
 
 	// an empty patch
-	createdNoxuInstance, err = noxuNamespacedResourceClient.Patch("foo", types.MergePatchType, []byte(`{}`), metav1.UpdateOptions{})
+	patchedNoxuInstance, err = noxuNamespacedResourceClient.Patch("foo", types.MergePatchType, []byte(`{}`), metav1.UpdateOptions{})
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	// an empty patch is a no-op patch. make sure it does not increment resourceVersion
+	if patchedNoxuInstance.GetResourceVersion() != resourceVersion {
+		t.Fatalf("unexpected error: resourceVersion changed on empty patch. expected %s, got %s", resourceVersion, patchedNoxuInstance.GetResourceVersion())
+	}
 
 	originalJSON, err := runtime.Encode(unstructured.UnstructuredJSONScheme, createdNoxuInstance)
```

Comment:
this is referencing `createdNoxuInstance` which is no longer the result of the operation on line 689, right? should it change to `patchedNoxuInstance`?

Reply:
fixed this.